ansible_collections/f5networks/f5_modules/plugins/modules/bigip_ssl_key_cert.py

```diff
@@ -0,0 +1,815 @@
+  returned: changed
+  type: bool
+  sample: true
+param2:
+  description: The new param2 value of the resource.
+  returned: changed
+  type: str
+  sample: Foo is bar
+'''
+
+import hashlib
+import os
+import re
+
+from ansible.module_utils.basic import (
+    AnsibleModule, env_fallback
+)
+
+from ..module_utils.bigip import F5RestClient
+from ..module_utils.common import (
+    F5ModuleError, AnsibleF5Parameters, transform_name, f5_argument_spec, fq_name
+)
+from ..module_utils.icontrol import (
+    TransactionContextManager, upload_file
+)
+
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
+
+
+def merge_two_dicts(x, y):
+    z = x.copy()
+    z.update(y)
+    return z
+
+
+class Parameters(AnsibleF5Parameters):
+    download_path = '/var/config/rest/downloads'
+
+    api_map = {
+        'sourcePath': 'source_path',
+        'issuerCert': 'issuer_cert',
+    }
+
+    api_attributes = [
+        'passphrase',
+        'sourcePath',
+        'issuerCert',
+    ]
+
+    returnables = [
+        'filename',
+        'checksum',
+        'source_path',
+        'issuer_cert',
+    ]
+
+    updatables = [
+        'checksum',
+        'key_checksum',
+        'cert_checksum',
+        'content',
+        'issuer_cert',
+        'source_path',
+    ]
+
+
+class ApiParameters(Parameters):
+    @property
+    def checksum(self):
+        if self._values['checksum'] is None:
+            return None
+        pattern = r'SHA1:\d+:(?P<value>[\w+]{40})'
+        matches = re.match(pattern, self._values['checksum'])
+        if matches:
+            return matches.group('value')
+        else:
+            return None
+
+    @property
+    def key_filename(self):
+        if self._values['name'] is None:
+            return None
+        if not self._values['name'].endswith('.key'):
+            return None
+        return self._values['name']
+
+    @property
+    def key_source_path(self):
+        if self.key_filename is None:
+            return None
+        if self._values['sourcePath'] is None:
+            return None
+        else:
+            return self._values['sourcePath']
+
+    @property
+    def cert_filename(self):
+        if self._values['name'] is None:
+            return None
+        if not self._values['name'].endswith('.crt'):
+            return None
+        return self._values['name']
+
+    @property
+    def cert_source_path(self):
+        if self.cert_filename is None:
+            return None
+        if self._values['sourcePath'] is None:
+            return None
+        else:
+            return self._values['sourcePath']
+
+    # @property
+    # def key_checksum(self):
+    #     if not self._values['name'].endswith('.key'):
+    #         return None
+    #     return self._values['checksum']
+
+    # @property
+    # def cert_checksum(self):
+    #     if not self._values['name'].endswith('.cert'):
+    #         return None
+    #     return self._values['checksum']
+
```

Comment:
why is this commented out ?

Reply:
The API Parameters don't have key and cert checksums that can be calculated on the fly. They need to be computed before. See line 742 and 763. Forgot to remove these comments.

Reply:
Actually, added them and added the checksum regex code.